addons/binding/org.openhab.binding.geofence/src/main/java/org/openhab/binding/geofence/internal/BindingComponent.java

```diff
@@ -0,0 +1,264 @@
+                            PointType location = locationProv.getLocation();
+                            String locationConfig = (String) properties.get(BindingConstants.CONFIG_LOCATION);
+                            if (location != null) {
+                                if (locationConfig == null || !location.toFullString().equals(locationConfig)) {
+                                    logger.debug("Updating location from system config: {}", location.toFullString());
+                                    properties.put(BindingConstants.CONFIG_LOCATION, location.toFullString());
+                                    configuration.update(properties);
+                                }
+                            } else if (locationConfig != null) {
+                                logger.debug("Clear primary location config as system location is missing");
+                                properties.remove(BindingConstants.CONFIG_NAME);
+                                properties.remove(BindingConstants.CONFIG_LOCATION);
+                                properties.put(BindingConstants.CONFIG_RADIUS, 100);
+                                configuration.update(properties);
+                            }
+                        }
+                    }
+                } catch (IOException e) {
+                    logger.error("Unable to update primary location from system config.", e);
+                }
+            }
+        }
+    }
+
+    private Dictionary<String, Object> copy(Dictionary<String, Object> properties) {
+        if (properties != null) {
+            Hashtable<String, Object> newProp = new Hashtable<>();
+
+            Enumeration<String> keys = properties.keys();
+            while (keys.hasMoreElements()) {
+                String key = keys.nextElement();
+                newProp.put(key, properties.get(key));
+            }
+
+            return newProp;
+        }
+        return null;
+    }
+
+    /**
+     * Deactivate the binding. It stops the HTTP callback endpoint and stops the device discovery service.
+     *
+     * @param componentContext Component context.
+     */
+    protected void deactivate(ComponentContext componentContext) {
+        logger.info("Deactivating Geofence Binding");
+        logger.debug("Stopping callback servlets");
+        otHTTPEndpoint.deactivate();
+        glHTTPEndpoint.deactivate();
+
+        logger.debug("Stopping discovery service");
+        this.serviceRegistration.unregister();
+
+        super.deactivate(componentContext);
+    }
+
+    @Reference(unbind = "unsetHttpService")
+    protected void setHttpService(HttpService httpService) {
+        this.httpService = httpService;
+    }
+
+    @SuppressWarnings("unused")
+    protected void unsetHttpService(HttpService httpService) {
+        this.httpService = null;
+    }
+
+    @Reference(unbind = "unsetThingRegistry")
+    protected void setThingRegistry(ThingRegistry thingRegistry) {
```

Comment:
Bindings are normally not allowed to access the ThingRegistry and I would claim that it isn't necessary in your case either.
As you know that you are only interested in handlers that you create yourself in this binding, you could simply keep them in a hashmap within this factory and answer the according questions from the servlets yourself (ideally by specifying a new interface with the required method, let the factory implement this interface and reach `this` to the servlets as a parameter of that interface).